infra/scripts/replace-running-components.sh

```diff
@@ -0,0 +1,173 @@
+#!/bin/bash
+# Copyright 2018 VMware, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+if [ -z "$PS1" ]; then
+    interactive=1
+else
+    interactive=0
+fi
+
+BASE_DIR=$(dirname $(readlink -f "$BASH_SOURCE"))
+VIC_DIR=$(dirname $(readlink -f $BASE_DIR/..))
+
+# Run the command given on the VCH instead of locally
+function on-vch() {
+    ssh -oUserKnownHostsFile=/dev/null -oStrictHostKeyChecking=no -i $VIC_KEY root@$VCH_IP -C $@ 2>/dev/null
+}
+
+function get-thumbprint() {
+    govc about.cert -k -thumbprint | awk '{print $NF}'
```

Comment:
might want `pipefail` and or $? check for govc.

```
$ govc about.cert -k -thumbprint | awk '{print $NF}'
govc: dial tcp 192.168.78.128:443: getsockopt: no route to host
```

Reply:
strictly speaking `govc ls`  is called first, but I think pipefail should just be on in the whole script, I was already thinking about that